src/transformStateWithClones.js

```diff
@@ -8,6 +8,23 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateHistory = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateHistory, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateHistory[key];
+      }
+    } else if (action.type === 'clear') {
+      stateHistory = {};
+    }
+    states.push({ ...stateHistory });
```

Comment:
The `states.push({ ...stateHistory });` line should be placed outside of the `if-else` block, at the end of each loop cycle. This ensures that the state is pushed after processing the current action, as per the additional prompt instructions.